src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -230,6 +240,8 @@ public ReadOnlySequence<T> Slice(int start, SequencePosition end)
         /// <param name="length">The length of the slice</param>
         public ReadOnlySequence<T> Slice(SequencePosition start, int length)
         {
+            if (length < 0)
+                ThrowHelper.ThrowArgumentValidationException(0);
```

Comment:
Why not pass in ExceptionArgument.length here directly instead of `0`?

Reply:
Because it should be start.

Reply:
Sorry, can we use the enum `ExceptionArgument.start` then?

Reply:
We send only one param (start) so if start is negative we throw about start else we throw about length. Here we should throw about length so we send correct start. It is trick code.

Reply:
It is aggressive in-lining code so to make it short we make only one call to throw all exceptions and we do not pass last param (length).

Reply:
Can you please add a comment here please, just so it is clear that passing in a value `>= 0` means throw on length being invalid?

Reply:
Also, why not use a bool instead? True means throw on start, false means throw on length?

Reply:
> It is aggressive in-lining code so to make it short we make only one call to throw all exceptions and we do not pass last param (length).

Using an enum (like in the other throw helpers) would also work here and still have aggressive in-lining, no?

Reply:
I copy trick from `private static Exception CreateArgumentValidationException<T>(ReadOnlySequenceSegment<T> startSegment, int startIndex, ReadOnlySequenceSegment<T> endSegment)` method but if have 3 params because the last endIndex do not passed.